### ISSUE TITLE
Test failed: tracing\eventpipe\providervalidation\providervalidation\providervalidation.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#26241]
**Job:**
https://helix.dot.net/api/2019-06-17/jobs/40a0c6d5-430e-4787-9f60-9317e3055a63/workitems/PayloadGroup0/console

**Failed tests:**
tracing\eventpipe\providervalidation\providervalidation\providervalidation.cmd

**Log:**
```
 tracing\eventpipe\providervalidation\providervalidation\providervalidation.cmd [FAIL]
      
      Return code:      1
      Raw output file:      C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Work\2cf345e3-c30f-4727-90b8-5c1641ee9d44\Exec\tracing\eventpipe\Reports\tracing.eventpipe\providervalidation\providervalidation\providervalidation.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Payload\corerun.exe" providervalidation.dll 
      TEST STARTING
      TEST PASSED!
      TEST STARTING
      Test FAILED!
      No events for provider "Microsoft-DotNETCore-SampleProfiler"
      Configuration:
      {
      bufferSize: 1024,
      providers: [
      MyEventSource:0xFFFFFFFFFFFFFFFF:5,
      Microsoft-DotNETCore-SampleProfiler:0xFFFFFFFFFFFFFFFF:5,
      SentinelEventSource:0xFFFFFFFFFFFFFFFF:5,
      ]
      }
      
      Expected:
      {
      "MyEventSource" = 1000 +- 300
      "Microsoft-Windows-DotNETRuntimeRundown" = -1 +- -0
      "Microsoft-DotNETCore-SampleProfiler" = -1 +- -0
      }
      
      Actual:
      {
      "MyEventSource" = 1000
      "Microsoft-DotNETCore-EventPipe" = 1
      "Microsoft-Windows-DotNETRuntimeRundown" = 1931
      }
      Expected: 100
      Actual: -1
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
      To run the test:
      > set CORE_ROOT=C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Payload
      > C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Work\2cf345e3-c30f-4727-90b8-5c1641ee9d44\Exec\tracing\eventpipe\providervalidation\providervalidation\providervalidation.cmd
      Expected: True
      Actual:   False
      Stack Trace:
        F:\workspace\_work\1\s\bin\tests\Windows_NT.x64.Checked\TestWrappers\tracing.eventpipe\tracing.eventpipe.XUnitWrapper.cs(431,0): at tracing_eventpipe._providervalidation_providervalidation_providervalidation_._providervalidation_providervalidation_providervalidation_cmd()
      Output:
        
        Return code:      1
        Raw output file:      C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Work\2cf345e3-c30f-4727-90b8-5c1641ee9d44\Exec\tracing\eventpipe\Reports\tracing.eventpipe\providervalidation\providervalidation\providervalidation.output.txt
        Raw output:
        BEGIN EXECUTION
         "C:\dotnetbuild\work\40a0c6d5-430e-4787-9f60-9317e3055a63\Payload\corerun.exe" providervalidation.dll 
        TEST STARTING
        TEST PASSED!
        TEST STARTING
        Test FAILED!
        No events for provider "Microsoft-DotNETCore-SampleProfiler"
        Configuration:
        {
        bufferSize: 1024,
        providers: [
        MyEventSource:0xFFFFFFFFFFFFFFFF:5,
        Microsoft-DotNETCore-SampleProfiler:0xFFFFFFFFFFFFFFFF:5,
        SentinelEventSource:0xFFFFFFFFFFFFFFFF:5,
        ]
        }
        
        Expected:
        {
        "MyEventSource" = 1000 +- 300
        "Microsoft-Windows-DotNETRuntimeRundown" = -1 +- -0
        "Microsoft-DotNETCore-SampleProfiler" = -1 +- -0
        }
        
        Actual:
        {
        "MyEventSource" = 1000
        "Microsoft-DotNETCore-EventPipe" = 1
        "Microsoft-Windows-DotNETRuntimeRundown" = 1931
        }
        Expected: 100
        Actual: -1
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
```

**Comment by josalem** [comment]
Will attempt to repro.  Happens with `JitStress` on.
```
set COMPlus_TieredCompilation=0
set COMPlus_JitStress=2
```

**Comment by josalem** [comment]
I'm unable to reproduce this on my local box, and am trying to secure a duplicate VM to the Helix queue this failed in.

**Comment by AriNuer** [comment]
Same test `tracing\\eventpipe\\providervalidation\\providervalidation\\providervalidation.cmd` failed with different failure here:
```

Error message
cmdLine:C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\providervalidation\\providervalidation\\providervalidation.cmd Timed Out
Return code: -100
Raw output file: C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\Reports\\tracing.eventpipe\\providervalidation\\providervalidation\\providervalidation.output.txt
Raw output:
BEGIN EXECUTION
1 file(s) copied.
" C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Payload\\crossgen.exe\" /Platform_Assemblies_Paths C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Payload;C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\providervalidation\\providervalidation\\IL;C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\providervalidation\\providervalidation /in providervalidation.org /out providervalidation.dll\r\nMicrosoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation. All rights reserved.
Native image providervalidation.dll generated successfully.
\"C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Payload\\corerun.exe\" providervalidation.dll 
TEST STARTING
TEST PASSED!
TEST STARTING
TEST PASSED!
cmdLine:C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\providervalidation\\providervalidation\\providervalidation.cmd Timed Out
Test Harness Exitcode is : -100
To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Payload
> C:\\dotnetbuild\\work\\af130a99-6ce1-4fad-8ac5-7831fb482f79\\Work\\b10d26dc-7133-4fcf-a792-85d3b498720e\\Exec\\tracing\\eventpipe\\providervalidation\\providervalidation\\providervalidation.cmd
Expected: True
Actual: False

Stack trace
at tracing_eventpipe._providervalidation_providervalidation_providervalidation_._providervalidation_providervalidation_providervalidation_cmd() in F:\workspace\_work\1\s\bin\tests\Windows_NT.x64.Release\TestWrappers\tracing.eventpipe\tracing.eventpipe.XUnitWrapper.cs:line 433
```
Details:
https://dnceng.visualstudio.com/internal/_build/results?buildId=318528&view=ms.vss-test-web.build-test-results-tab&runId=9218856&resultId=110424&paneView=debug

**Comment by jkotas** [comment]
Timeout hit again in https://github.com/dotnet/coreclr/pull/27299

**Comment by josalem** [comment]
@jkoritzinsky, can I/how do I go about getting a dump from the timeout Jan linked to above?

**Comment by jkoritzinsky** [comment]
You can go to the run details on Azure DevOps. Then follow these steps from Arcade to get the dump: https://github.com/dotnet/arcade/blob/master/Documentation/Dumps/Dumps.md

**Comment by josalem** [comment]
This issue has been root caused and a fix has been put out in the appropriate place for it.  There is one last change that needs to be made to the test to make it resilient to timing out and to update the TraceEvent version once that is fixed.  Those changes will have to be done after repo consolidation at this point.

**Comment by safern** [comment]
@josalem I'm hitting this in the runtime repo on: https://helix.dot.net/api/2019-06-17/jobs/af5c83cb-842d-43c3-8a86-52a4a23f7f33/workitems/PayloadGroup0/console

**Comment by josalem** [comment]
Fixed in Microsoft/PerfView#1047.  Looks like we never came back to upgrade the TraceEvent version (needs to be >= 2.0.48 to have the fix).  Should be a simple fix at this point.
